data/sea/54-zebra.h

```diff
@@ -155,73 +142,181 @@ static ierror_msg_t zebra_translate
-                uint8_t* bytes = anemone_mempool_alloc(mempool, count + 1);
-                memcpy(bytes, data.b + elem_start, count);
-                bytes[count] = '\0';
-                *dst = bytes;
-                return 1;
-            }
+    const zebra_column_tag_t tag = src->tag;
+    const zebra_column_variant_t data = src->of;
 
-        case ZEBRA_INT:
-            *dst = data.i + elem_start;
+    switch (tag) {
+        case ZEBRA_COLUMN_UNIT: {
+            return 0;
+        }
+
+        case ZEBRA_COLUMN_INT: {
+            *dst = data._int.values + elem_start;
             return 1;
+        }
 
-        case ZEBRA_DOUBLE:
-            *dst = data.d + elem_start;
+        case ZEBRA_COLUMN_DOUBLE: {
+            *dst = data._double.values + elem_start;
             return 1;
+        }
 
-        case ZEBRA_ARRAY:
-            {
-                istring_t *strings = anemone_mempool_alloc (mempool, count * sizeof (istring_t));
-                int offset = 1;
-                for (int i = 0; i != count; ++i) {
-                    int64_t index = i + elem_start;
-                    int64_t start = data.a.n[index]   - data.a.n[0];
-                    int64_t len   = data.a.n[index+1] - data.a.n[index];
-                    offset = zebra_translate_table (mempool, start, len, (void**)(strings + i), &data.a.table);
-                }
-                *dst = strings;
-                return offset;
+        case ZEBRA_COLUMN_ENUM: {
+            int64_t column_count = data._enum.columns.count;
+            int64_t offset = zebra_translate_column_tags
+                ( column_count
+                , dst
+                , data._enum.tags );
+            for (int64_t i = 0; i != column_count; ++i) {
+                offset += zebra_translate_column
+                    ( mempool
+                    , elem_start
+                    , elem_count
+                    , dst + offset
+                    , data._enum.columns.columns + i );
             }
+            return offset;
+        }
+
+        case ZEBRA_COLUMN_STRUCT: {
+            int64_t offset = 0;
+            int64_t column_count = data._struct.columns.count;
+            for (int64_t i = 0; i != column_count; ++i) {
+                offset += zebra_translate_column
+                    ( mempool
+                    , elem_start
+                    , elem_count
+                    , dst + offset
+                    , data._struct.columns.columns + i );
+            }
+            return offset;
+        }
+
+        case ZEBRA_COLUMN_NESTED: {
+            return zebra_translate_column_nested
+                ( mempool
+                , elem_start
+                , elem_count
+                , dst
+                , data._nested.indices
+                , &data._nested.table );
+        }
+
+        case ZEBRA_COLUMN_REVERSED: {
+            fprintf (stderr, "Fatal error: encountered a reversed Zebra column.\n");
```

Comment:
afaik Reversed was meant for the merge, when Icicle gets the entity it should be in the reversed order already